x-pack/plugins/dashboard_enhanced/public/services/drilldowns/abstract_dashboard_drilldown/abstract_dashboard_drilldown.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { DataPublicPluginStart } from 'src/plugins/data/public';
+import { DashboardStart } from 'src/plugins/dashboard/public';
+import { reactToUiComponent } from '../../../../../../../src/plugins/kibana_react/public';
+import {
+  TriggerId,
+  TriggerContextMapping,
+} from '../../../../../../../src/plugins/ui_actions/public';
+import { CollectConfigContainer } from './components';
+import {
+  UiActionsEnhancedDrilldownDefinition as Drilldown,
+  UiActionsEnhancedBaseActionFactoryContext as BaseActionFactoryContext,
+  AdvancedUiActionsStart,
+} from '../../../../../ui_actions_enhanced/public';
+import { txtGoToDashboard } from './i18n';
+import {
+  StartServicesGetter,
+  CollectConfigProps,
+} from '../../../../../../../src/plugins/kibana_utils/public';
+import { KibanaURL } from '../../../../../../../src/plugins/share/public';
+import { Config } from './types';
+
+export interface Params {
+  start: StartServicesGetter<{
+    uiActionsEnhanced: AdvancedUiActionsStart;
+    data: DataPublicPluginStart;
+    dashboard: DashboardStart;
+  }>;
+}
+
+export abstract class AbstractDashboardDrilldown<T extends TriggerId>
+  implements Drilldown<Config, T, BaseActionFactoryContext<T>> {
+  constructor(protected readonly params: Params) {}
+
+  public abstract readonly id: string;
+
+  public abstract readonly supportedTriggers: () => T[];
+
+  protected abstract getURL(config: Config, context: TriggerContextMapping[T]): Promise<KibanaURL>;
+
+  public readonly order = 100;
+
+  public readonly getDisplayName = () => txtGoToDashboard;
+
+  public readonly euiIcon = 'dashboardApp';
+
+  private readonly ReactCollectConfig: React.FC<
+    CollectConfigProps<Config, BaseActionFactoryContext<T>>
+  > = (props) => <CollectConfigContainer {...props} params={this.params} />;
+
+  public readonly CollectConfig = reactToUiComponent(this.ReactCollectConfig);
+
+  public readonly createConfig = () => ({
+    dashboardId: '',
+    useCurrentFilters: true,
+    useCurrentDateRange: true,
+  });
+
+  public readonly isConfigValid = (config: Config): config is Config => {
+    if (!config.dashboardId) return false;
+    return true;
+  };
+
+  public readonly getHref = async (
+    config: Config,
+    context: TriggerContextMapping[T]
+  ): Promise<string> => {
+    const url = await this.getURL(config, context);
+    return url.path;
+  };
+
+  public readonly execute = async (config: Config, context: TriggerContextMapping[T]) => {
+    const url = await this.getURL(config, context);
+    await this.params.start().core.application.navigateToApp(url.appName, { path: url.appPath });
```

Comment:
Btw, there is also `navigateToUrl` method. 
I think we can and should use it. Then we don't need URL parsing and can remove temporary `KibanaURL` implementation. 

```
this.params.start().core.application.navigateToUrl(await this.getURL(config, context))
```

Reply:
Yes, good catch, I've also seen it while refactoring this, but decided to leave as is for now so to not accidentally break anything.